src/EPR.SubsidiaryBulkUpload.Application/Services/CsvProcessor.cs

```diff
@@ -31,5 +20,22 @@ public async Task<IEnumerable<T>> ProcessStream<T>(Stream stream, IReaderConfigu
 
             return records;
         }
+
+        public async Task<IEnumerable<TD>> ProcessStreamWithMapping<TD, TM>(Stream stream, IReaderConfiguration configuration)
+            where TM : ClassMap
+        {
+            try
+            {
+                var (_, theList) = _parserClass.ParseWithHelper(stream, configuration);
+                return (IEnumerable<TD>)theList;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error occurred while processing ProcessStream {Error}", ex.Message);
+                throw;
```

Comment:
Why is this thrown here?
What is the exceptio0n caught, are there specialised ones than Exception?
If this is really needed (although I suggest it isn't), make a specialised exception instead of just throwing the same Exception.
Oh and this will be a SonarQube code smell!

Reply:
I am addressing the outer expection into the error processing side. this one is the only one left. rest of exceptions are removed.